### PR TITLE
Adding config prop to CrossmintPayButton

### PR DESCRIPTION
**Pull request by pgarciaegido**
Here I add the config type to the CrossmintPayButton, and refactor a bit, incuding interfaces to the `types.ts` file.

### CODE REVIEW

File: packages/ui/react-ui/src/types.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    clientName = "X-Client-Name",
+}
+
+enum theme {
+    light = "light",
+    dark = "dark",
+}
+
+export enum payButtonTypes {
+    solanaCandyMachine = "solana-candy-machine",
```
Comment (mPaella) [inline packages/ui/react-ui/src/types.ts:23]: In general we should follow [this](https://github.com/Paella-Labs/crossbit-main/blob/dc6f1bf0b25d6a819f542ad630027f49d1be4b70/models/ClientIntegration.ts#L7) enum

File: packages/ui/react-ui/src/types.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+export interface BaseButtonProps {
+    className?: string;
+    disabled?: boolean;
+    onClick?: (e: MouseEvent<HTMLButtonElement>) => void;
+    style?: CSSProperties;
+    tabIndex?: number;
+    clientId: string;
+    auctionId?: string;
+    theme?: theme;
+    development: boolean;
```
Comment (mPaella) [inline packages/ui/react-ui/src/types.ts:39]: should `development` be optional? default to false

File: packages/ui/react-ui/src/types.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+export interface CrossmintPayButtonProps extends BaseButtonProps {
+    collectionTitle?: string;
+    collectionDescription?: string;
+    collectionPhoto?: string;
+    mintTo?: string;
+    emailTo?: string;
+    listingId?: string;
+    showOverlay?: boolean;
+    hideMintOnInactiveClient?: boolean;
+    config?: PayButtonConfig;
```
Comment (mPaella) [inline packages/ui/react-ui/src/types.ts:51]: Maybe a different name to be more specific that this is related to their blockchain contract? `mintConfig` or something similar

Comment (pgarciaegido) [reply packages/ui/react-ui/src/types.ts:53]: That makes sense. wdyt @alfonso-paella ?

Comment (alfonso-paella) [reply packages/ui/react-ui/src/types.ts:53]: Sure

File: packages/ui/react-ui/src/types.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    dark = "dark",
+}
+
+export enum mintingContractTypes {
+    CANDY_MACHINE = "candy-machine",
+    SOLANA_AUCTION = "solana-auction",
+    ERC_721 = "erc-721",
+}
+
+export interface PayButtonConfig {
```
Comment (alfonso-paella) [inline packages/ui/react-ui/src/types.ts:28]: On discord we decided to not have any client side validation / enforcement of what goes on the mintConfig property, so we can release new contracts without doing a new SDK release each time

As a result,, this type is not needed and should be replaced with an "any"

**Review by alfonso-paella — CHANGES_REQUESTED** [review]

**Review by alfonso-paella — APPROVED** [review]

**Pull request merged**